pycloudlib/instance.py

```diff
@@ -369,7 +369,32 @@ def _wait_for_cloudinit(self, *, raise_on_failure: bool):
             then this method will raise an `OSError`.
         """
         self._log.debug('_wait_for_cloudinit to complete')
-        has_wait = "--wait" in self.execute("cloud-init status --help").stdout
+        test_instance_command = "whoami"
+
+        # Checks first if we can reach the instance with a basic command
+        result = self.execute(test_instance_command)
+        if result.failed:
+            retries = 10
+            while retries:
+                result = self.execute(test_instance_command)
+
+                if result.ok:
+                    break
+
+                retries -= 1
+                time.sleep(10)
+
+            if result.failed:
+                raise OSError(
+                    "{}\n{}".format(
+                        "Instance can't be reached",
+                        "Failed to execute {} command".format(
+                            test_instance_command)
+                    )
+                )
```

Comment:
Thanks for this update, this looks correct to me now. This feels like it could quite naturally be refactored to a separate method now: `_wait_for_execute` or something along those lines. And it might make sense to call it from `BaseInstance.wait` instead of from `BaseInstance._wait_for_cloudinit` as it's not particularly related to cloud-init any longer?